test/include/graph_test/graph_test.h

```diff
@@ -32,6 +37,35 @@ class EmptyDBTest : public PrivateGraphTest {
     std::string getInputDir() override { KU_UNREACHABLE; }
 };
 
+class ConcurrentTest {
+public:
+    ConcurrentTest(bool& connectionsPaused, main::Connection& connection, std::string databasePath):
+      connectionPaused{connectionsPaused}, connection{connection}, databasePath{databasePath} {}
+    void execute() {
+        connThread = std::thread(&ConcurrentTest::runStatements, this);
+    }
+    void join() {
```

Comment:
Looks to me `join()` is unnecessary, I think you should always join inside `execute`, which makes the interface simpler.